actionmailer/test/message_delivery_test.rb

```diff
@@ -101,4 +108,46 @@ def test_should_enqueue_and_run_correctly_in_activejob
       @mail.deliver_later
     end
   end
+
+  test 'job delegates error handling to mailer' do
+    # Superclass not rescued by mailer's rescue_from RuntimeError
+    message = DelayedMailer.test_raise('StandardError')
+    assert_raise(StandardError) { message.deliver_later }
+    assert_nil DelayedMailer.last_error
+    assert_nil DelayedMailer.last_rescue_from_instance
+
+    # Rescued by mailer's rescue_from RuntimeError
+    message = DelayedMailer.test_raise('DelayedMailerError')
+    assert_nothing_raised { message.deliver_later }
+    assert_equal 'boom', DelayedMailer.last_error.message
+    assert_kind_of DelayedMailer, DelayedMailer.last_rescue_from_instance
+  end
+
+  class DeserializationErrorFixture
+    include GlobalID::Identification
+
+    def self.find(id)
+      raise 'boom, missing find'
+    end
+
+    attr_reader :id
+    def initialize(id = 1)
+      @id = id
+    end
+
+    def to_global_id(options = {})
+      super app: 'foo'
+    end
+  end
+
+  test 'job delegates deserialization errors to mailer class' do
+    # Inject an argument that can't be deserialized.
+    message = DelayedMailer.test_message(DeserializationErrorFixture.new)
+
+    # DeserializationError is raised, rescued, and delegated to the handler
+    # on the mailer class.
+    assert_nothing_raised { message.deliver_later }
+    assert_equal DelayedMailer, DelayedMailer.last_rescue_from_instance
+    assert_equal 'Error while trying to deserialize arguments: boom, missing find', DelayedMailer.last_error.message
+  end
```

Comment:
Demonstrating that AJ deserialization errors are handled by a rescue_from on DelayedMailer.